Sapflow_prelim_analysis.R

```diff
@@ -0,0 +1,88 @@
+# Sapflow Initial Analysis
+
+#load in packages(there's probably a way to do this all at once)
+library(readr) 
+library(dplyr) 
+library(tidyr) 
+library(ggplot2)
+library(lubridate)
+
+#load in sapflow data (already bound into single csv from 
+#create-time-series.R script) 
+
+readRDS("sapflow.rds") -> sapflow_data
+
+#start and end times for data
+DATA_BEGIN <- as_datetime("2024-04-17 00:00:00", tz = "EST")
+DATA_END <- as_datetime("2024-04-30 23:59:59", tz = "EST")
+
+#These values are from 2022's TEMPEST day 1 event
+#EVENT_START <- as_datetime("2022-06-22 05:30:00", tz = "EST")
+#EVENT_STOP <- as_datetime("2022-06-22 14:30:00", tz = "EST")
+
+
+#tidy up data
+#grepl!! Steph you're a genius!! I was doing it the long way!!
+
+sapflow_data <- sapflow_data %>% 
+  mutate(Value = as.numeric(Value),
+         Date = date(TIMESTAMP),
+         Hour = hour(TIMESTAMP)) %>% 
+  filter(TIMESTAMP >= DATA_BEGIN, TIMESTAMP <= DATA_END, 
+         Value > 0.02, Value <2, !grepl("D", ID)) 
+
+#Add species column to sapflow_data
+sapflow_data <- merge(species, sapflow_data, by.x = "ID", by.y = "Sensor_ID", all.x = TRUE)
+
+sapflow_data %>% 
+  drop_na(ID) %>%
+  select(TIMESTAMP, Date, Hour, ID, Species, Plot.y, Location, Value) %>% 
+  mutate(plot = substr(ID,1,1),
+         plot = case_when(plot == "C" ~ "Control",
+                          plot == "S" ~ "Saltwater",
+                          plot == "F" ~ "Freshwater")) -> sf_dat
+
+#From Steph: For dTmax, we're using the max sapflow method, which needs to meet the following criteria:
+# (1) Be between the hours of midnight and 5am
+# (2) Be the maximum sapflow value
+#Filter accordingly: 
+
+sf_dat %>% 
+  filter(Hour >= 0, Hour <= 5) %>% 
+  group_by(Date, Plot.y, Species, ID) %>% 
+  summarise(dTmax = max(Value, na.rm = TRUE), dTmax_Timestamp = TIMESTAMP[which.max(Value)]) -> sapflow_dtmax
+
+#Graph sapflow data with dtmax calculation to double check correct calculation: 
+sf_dat %>% 
```

Comment:
Is Plot.y the same as Plot.x? If so it would be much cleaner to  include Plot in your joining statement and just have one Plot column.